boilerplate/app/components/Toggle.tsx

```diff
@@ -0,0 +1,493 @@
+import React, { ComponentType, FC, useMemo, useState } from "react"
+import {
+  GestureResponderEvent,
+  Image,
+  ImageStyle,
+  StyleProp,
+  SwitchProps,
+  TextInputProps,
+  TextStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  View,
+  ViewStyle,
+} from "react-native"
+import Animated, { useAnimatedStyle, withTiming } from "react-native-reanimated"
+import { colors, spacing } from "../theme"
+import { iconRegistry } from "./Icon"
+import { Text, TextProps } from "./Text"
+
+type Variants = "checkbox" | "switch" | "radio"
+
+export interface ToggleProps extends Omit<TouchableOpacityProps, "style"> {
+  /**
+   * The variant of the toggle.
+   * Options: "checkbox", "switch", "radio"
+   * Default: "checkbox"
+   */
+  variant?: Variants
+  /**
+   * A style modifier for different input states.
+   */
+  status?: "error" | "disabled"
+  /**
+   * If false, input is not editable. The default value is true.
+   */
+  editable?: TextInputProps["editable"]
```

Comment:
Adds an `editable` prop to be inline with `TextField`'s editable prop. This is an alternative way to specific `status="disabled"` and does the same thing.